File: 7kyu/Complementary DNA/dna_strand.py
# https://www.codewars.com/kata/554e4a2f232cdd87d9000038/python
def DNA_strand(dna:str) -> str:
    complements = []
    for ch in dna:
        if ch == 'A':
            complements.append('T')
        elif ch == 'T':
            complements.append('A')
        elif ch == 'C':
            complements.append('G')
        elif ch == 'G':
            complements.append('C')
            
    return ''.join(complements)


# translate 기능이 있는 건 처음 알았다.
# import string
# def DNA_strand(dna):
#     return dna.translate(string.maketrans("ATCG","TAGC"))
            

File: 7kyu/Complementary DNA/DnaStrand.java
// https://www.codewars.com/kata/554e4a2f232cdd87d9000038/java

public class DnaStrand {
    public static String makeComplement(String dna) {
      char[] chars = dna.toCharArray();
      String complements = "";
  
      for (char ch: chars) {
          switch(ch){
              case 'A':
                complements += 'T';
                break;
              case 'T':
                complements += 'A';
                break;
              case 'C':
                complements += 'G';
                break;
              case 'G':
                complements += 'C';
                break;
          }
      }
      
      return complements;
    }
  }